draft-ietf-quic-transport.md

```diff
@@ -1746,7 +1747,8 @@ validate a client address.  Tokens are not integrated into the cryptographic
 handshake and so they are not authenticated.  For instance, a client might be
 able to reuse a token.  To avoid attacks that exploit this property, a server
 can limit its use of tokens to only the information needed to validate client
-addresses.
+addresses.  A server MUST NOT require the client to correlate any additional
+information with a token (e.g. a NewSessionTicket).
```

Comment:
```suggestion

Clients MAY use tokens obtained from the this version of QUIC with any connection attempt for the same version.   Clients do not need to consider other properties of the connection that is being attempted when selecting a token, including the choice of possible application protocols, session tickets, or other connection properties.
```

I don't think that this is a testable "MUST NOT".  It's more or less axiomatic.  A server can't prevent clients from attempting to use tokens.

Reply:
You're right that this isn't a testable "MUST NOT". I think the proposed language here is better suited with the language above describing a client's behavior.

Here I'm trying to provide some advice to server implementers that clients aren't going to keep track of which token came from where (or goes with which ticket), so if they expect that a token and ticket used on a new connection came from the same old connection, they're going to have a bad time.

Reply:
@nharper where are we at here?  I was suggesting a new paragraph, but I didn't get any sense from your reply about whether you thought that was necessary/useful/not.

Reply:
Sorry for the delay. The new paragraph (framed as client considerations) sounds like a better way to describe this point. I've added a version of that paragraph and rebased this PR. (There was a merge conflict for the change above.)